Sources/Profile-Actions.php

```diff
@@ -969,4 +969,196 @@ function subscriptions($memID)
 		$context['sub_template'] = 'user_subscription';
 }
 
+/**
+ * Function to allow the user to todo Privacy stuff
+ *
+ * @param int $memID The ID of the member
+ */
+function getProfileData($memID)
+{
+	global $txt, $user_profile, $context, $smcFunc, $user_info;
+
+	if ($memID == $user_info['id'])
+		$context['pdc']['own'] = true;
+	else
+		$context['pdc']['name'] = $context['member']['name'];
+	
+	if (!empty($_REQUEST['nofound']))
+		$context['pdc']['nofound'] = true;
+	
+	if (empty($_REQUEST['activity']))
+		return;
+	
+	$mode = $_REQUEST['activity'];
+
+	$profileData = array();
+	if ($mode == 'profile') // profile
+	{
+		loadMemberData($memID, false, 'profile');
+		$profile = $user_profile[$memID];
+		$removeFields = array('secret_question','tfa_secret','password_salt');
+		foreach ($removeFields as $value)
+		{
+			unset($profile[$value]);
+		}
+		foreach($profile as $key => &$value)
+		{
+			if (is_array($value))
+				$value = $smcFunc['json_encode']($value);
+		}
+		$profileData[0] = array_keys($profile);
+		$profileData[1] = $profile;
+		call_integration_hook('integrate_getProfile_profile', array(&$profileData));
+	}
+	elseif ($mode == 'messages') // messages
```

Comment:
100k messages takes 500 ms -> looks good to me
the file stuff got many weakness 
it created by his self a new gdpr topic
it could be infected by evil software
it take more time to proceed -> the cpu load is overall higher

Reply:
How much memory was used to process those 100,000 messages all at once?

Reply:
@Sesquipedalian That's very important indeed, especially problematic on more aggressively resource-limited shared hosting. Plus hosting many boards on a server executing that kind of queries can become a nuisance. Always assume low resources is probably a good idea.

Reply:
It needs to be realistic,
100k message  is more than sm.org got
And you will not host setup like this on freehost

Reply:
That's not an answer to the question, @albertlast. 🤨

There are very good reasons why SMF uses paging to perform any operation involving large numbers of table rows. Those reasons apply here just as much as they do anywhere else.

Reply:
you didn't awnser the question for the xml solution.

I test some solution,
handle the size better.